rllib/examples/env/multi_agent.py

```diff
@@ -87,6 +88,62 @@ def step(self, action_dict):
         return obs, rew, done, info
 
 
+class FlexAgentsMultiAgent(MultiAgentEnv):
+    """Env of independent agents, each of which exits after n steps."""
+
+    def __init__(self):
+        self.agents = {}
+        self.agentID = 0
+        self.dones = set()
+        self.observation_space = gym.spaces.Discrete(2)
```

Comment:
This shouldn't be hardcoded right?

Reply:
This code was copied from my repro -- idea was just to have a setting where agents are periodically added/deleted. You could make the obs space not hard coded. As for the agentID, that's a counter that is incremented whenever a new agent is spawned such that each agent always has a unique key